### ISSUE TITLE
Add a getTimeOrderedEpoch with instant parameter

### ISSUE THREAD

**Issue by nicolarosada** [f4b6a3/uuid-creator#82]
Hi,

It would be good to have UUID V7 creator getting an instant as parameter in order to create UUID at any timestamp.

Something like:
```
Instant aMinuteAgo = Instant.now().minusSeconds(60);
UUID uuid = UuidCreator.getTimeOrderedEpoch(aMinuteAgo);
```

The `getTimeOrderedEpoch()` is composed by 3 parts: time, counter and random. I understand that is impossible having the counter without maintaining a data structure with the past occurrences. Having the method using time and random only would be great anyway. We could optionally ask the user to provide the counter as input parameter with default 0.

Please let me know if this is something that you might be interested in developing. I can also help with a pull request in case.

**Comment by fabiolimace** [comment]
Hi @nicolarosada !

I agree that it would be nice to have this method.

It is possible to implement with the current UUIDv7 factory, but using [`Clock` instance](https://github.com/f4b6a3/uuid-creator/blob/3b6471642e95ca91901ce5efe817537aa0d1068b/src/main/java/com/github/f4b6a3/uuid/factory/AbstCombFactory.java#L65) is not convenient as 3 methods need to be overridden:

```java
    public static UUID getTimeOrderedEpoch(Instant instant) {
        Clock clock = new Clock() {
            
            @Override
            public Clock withZone(ZoneId zone) {
                // TODO Auto-generated method stub
                return null;
            }
            
            @Override
            public Instant instant() {
                return instant;
            }

            @Override
            public ZoneId getZone() {
                return null;
            }
        };
        return TimeOrderedEpochFactory.builder().withClock(clock).build().create();
    }
```

It could also be implemented by copying the logic of [`GUID.v7()`](https://github.com/f4b6a3/uuid-creator/blob/3b6471642e95ca91901ce5efe817537aa0d1068b/src/main/java/com/github/f4b6a3/uuid/alt/GUID.java#L321)

```java
    public static UUID getTimeOrderedEpoch_(Instant instant) {
        final long time = instant.toEpochMilli();
        ThreadLocalRandom random = ThreadLocalRandom.current();
        final long msb = (time << 16) | (random.nextLong() & 0xffL);
        final long lsb = random.nextLong();
        return new UUID(msb, lsb); // TODO: add version number: 7 
    }
```
Another approach would be to completelly refactor the UUIDv7 factory to use the same logic as the [ULID factory](https://github.com/f4b6a3/ulid-creator/blob/master/src/main/java/com/github/f4b6a3/ulid/UlidFactory.java) from ULID Creator.

The optional counter parameter is another issue to think about.

I still don't know what the best approach is or when I can do it.

**Comment by nicolarosada** [comment]
Thanks Fabio for your quick response, very appreciated.

What about going for the copy of `GUID.v7()` logic? it could be refactored later if other use cases arise.

**Comment by fabiolimace** [comment]
Approved and merged your PR. The best solution is the one you implemented. Thanks!

**Comment by nicolarosada** [comment]
Thank you very much!

**Comment by fabiolimace** [comment]
Released v5.3.3! 🎉
